tests/contract.rs

```diff
@@ -153,3 +158,49 @@ fn add_file(cache: &mut FileResolver, path: &Path, target: Target) -> io::Result
 
     Ok(filename.to_string())
 }
+
+fn check_diagnostics(path: &Path, ns: &Namespace) {
+    let mut expected = "// ----\n".to_owned();
+
+    for diag in ns.diagnostics.iter() {
+        if diag.level == Level::Warning || diag.level == Level::Error {
+            expected = format!(
+                "{}// {} ({}-{}): {}\n",
+                expected,
+                diag.level,
+                diag.loc.start(),
+                diag.loc.end(),
+                diag.message
+            );
+
+            for note in &diag.notes {
+                expected = format!(
+                    "{}// \tnote ({}-{}): {}\n",
+                    expected,
+                    note.loc.start(),
+                    note.loc.end(),
+                    note.message
+                );
+            }
+        }
+    }
+
+    let mut found = String::new();
+
+    let file = File::open(path).unwrap();
+
+    for line in BufReader::new(file).lines() {
+        let line = line.unwrap();
+
+        if found.is_empty() && !line.starts_with("// ----") {
+            continue;
+        }
+
+        found.push_str(&line);
+        found.push('\n');
+    }
+
+    if !found.is_empty() {
```

Comment:
I wonder if this should be an `assert!(!found.is_empty())` instead? It would force the developer of the test to think about it

Reply:
Totally agree, I've changed this as suggested